src/modules/navigator/rtl_direct_mission_land.cpp

```diff
@@ -212,6 +212,12 @@ void RtlDirectMissionLand::setActiveMissionItems()
 		}
 
 		mission_item_to_position_setpoint(_mission_item, &pos_sp_triplet->current);
+
+		// prevent lateral guidance from loitering at a waypoint as part of a mission landing if the altitude
+		// is not achieved.
+		if (_mission_item.nav_cmd == NAV_CMD_WAYPOINT) {
```

Comment:
@KonradRudin Was wondering if I would not need to add a check here as well if we are already on the mission landing part.

Reply:
Yes you actually would need that. When you are below the waypoint, you would anyways need to climb first then it is not important, but when you are way above it, it would not loiter before going to the descend route.